### PR TITLE
Added a Jenkinsfile, and fixed some buglets on Windows

### PR DESCRIPTION
**Pull request by jglick**
- [X] add something so that organization folder scanning knows about it
- [x] make sure tests pass on a new baseline
- [x] make sure tests pass on Windows

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -63,7 +63,10 @@
     </pluginRepositories>
     <properties>
         <jenkins.version>1.642.3</jenkins.version>
+        <jenkins-test-harness.version>2.19</jenkins-test-harness.version>
         <workflow-cps-plugin.version>2.28</workflow-cps-plugin.version>
+        <git-plugin.version>3.1.0</git-plugin.version>
```
Comment (svanoort) [inline pom.xml]: I really hate the use of a property here -- if you just move the normal and test deps adjacent it's trivial to bump both.  Use search+replace if you sometimes forget one.  This level of DRY is silly.

File: src/main/java/org/jenkinsci/plugins/workflow/cps/global/UserDefinedGlobalVariable.java [hunk 2]
```diff
@@ -69,7 +69,10 @@ public Object getValue(@Nonnull CpsScript script) throws Exception {
     public @CheckForNull String getHelpHtml() throws IOException {
         if (!help.exists())     return null;
 
-        return Jenkins.getActiveInstance().getMarkupFormatter().translate(FileUtils.readFileToString(help, Charsets.UTF_8));
+        return Jenkins.getActiveInstance().getMarkupFormatter().translate(
```
Comment (svanoort) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/global/UserDefinedGlobalVariable.java]: Noice.

**Comment by svanoort** [comment]
Needs to mention that you made a number of other small fixes in the title.

File: src/test/java/org/jenkinsci/plugins/workflow/libs/LibraryAdderTest.java [hunk 3]
```diff
@@ -290,7 +292,7 @@ public static void register(Object o) {
             clearInvocationCaches.invoke(metaClass);
         }
         for (WeakReference<ClassLoader> loaderRef : LOADERS) {
-            MemoryAssert.assertGC(loaderRef);
+            MemoryAssert.assertGC(loaderRef, false);
```
Comment (svanoort) [inline src/test/java/org/jenkinsci/plugins/workflow/libs/LibraryAdderTest.java]: Aha!

Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/libs/LibraryAdderTest.java]: Picked up from `workflow-cps` and verifies that there are no soft, much less hard, leaks.

**Pull request merged**